### PR TITLE
Supporting python 3.10 and dropping support for 3.6

### LINKED ISSUE #939: Support for python 3.10
What the title says. Python 3.10 has been out for ~10 months and support for it would be nice.

### LINKED ISSUE #1177: Drop support for python 3.6
### Problem Description

Python 3.6 has reached its EOL over a year ago and should be dropped.

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #939 
resolves #1177

### CODE REVIEW

File: sdv/tabular/ctgan.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """Wrapper around CTGAN model."""
 
 import numpy as np
-from ctgan import CTGANSynthesizer, TVAESynthesizer
+from ctgan import CTGAN, TVAE
```
Comment (pvk-developer) [inline sdv/tabular/ctgan.py:4]: I think that it would be better to just import `ctgan` and use `ctgan.CTGAN` and `ctgan.TVAE` in the code, just to avoid messing up the names.
eg:
`from sdv.tabular.ctgan import CTGAN` it becomes a bit unclear whether it will import the `CTGAN` class from `ctgan` or the `CTGAN` class declared in here.

**Review by pvk-developer — CHANGES_REQUESTED** [review]

File: tests/unit/tabular/test_copulas.py [hunk 2]
```diff
@@ -282,20 +282,20 @@ def test__fit_with_transformed_columns(self, gm_mock):
...
-            'a.value': [1, 2, 3]
+            'a': [1, 2, 3]
         })
         out = GaussianCopula._fit(gaussian_copula, data)
 
         # asserts
         assert out is None
         assert gaussian_copula._field_distributions == {
-            'a': 'a_distribution', 'a.value': 'a_distribution'}
+            'a': 'a_distribution', 'a': 'a_distribution'}
```
Comment (fealho) [inline tests/unit/tabular/test_copulas.py:292]: only need one

File: tests/unit/tabular/test_copulas.py [hunk 3]
```diff
@@ -282,20 +282,20 @@ def test__fit_with_transformed_columns(self, gm_mock):
...
         gm_mock.assert_called_once_with(
-            distribution={'a': 'a_distribution', 'a.value': 'a_distribution'})
+            distribution={'a': 'a_distribution', 'a': 'a_distribution'})
```
Comment (fealho) [inline tests/unit/tabular/test_copulas.py:294]: same

File: setup.cfg [hunk 4]
```diff
@@ -24,10 +24,6 @@ replace = version='{new_version}'
 search = __version__ = '{current_version}'
 replace = __version__ = '{new_version}'
 
-[bumpversion:file:conda/meta.yaml]
```
Comment (fealho) [inline setup.cfg:27]: Why was this removed?

**Review by fealho — APPROVED** [review]
Looks good 👍

File: setup.cfg (hunk 4 above)
Comment (amontanez24) [reply setup.cfg:27]: we deleted the conda folder a while ago and didn't update this

**Review by pvk-developer — APPROVED** [review]
